osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSlider.cs

```diff
@@ -43,25 +43,34 @@ public DrawableSlider(Slider s)
             Container<DrawableSliderTick> ticks;
             Container<DrawableRepeatPoint> repeatPoints;
 
-            InternalChildren = new Drawable[]
+            InternalChildren = new[]
             {
                 Body = new SnakingSliderBody(s)
                 {
                     PathWidth = s.Scale * 64,
                 },
                 ticks = new Container<DrawableSliderTick> { RelativeSizeAxes = Axes.Both },
                 repeatPoints = new Container<DrawableRepeatPoint> { RelativeSizeAxes = Axes.Both },
+                HeadCircle = new DrawableSliderHead(s, s.HeadCircle)
+                {
+                    OnShake = Shake
+                },
                 Ball = new SliderBall(s, this)
                 {
                     BypassAutoSizeAxes = Axes.Both,
                     Scale = new Vector2(s.Scale),
                     AlwaysPresent = true,
-                    Alpha = 0
-                },
-                HeadCircle = new DrawableSliderHead(s, s.HeadCircle)
-                {
-                    OnShake = Shake
+                    Alpha = 0,
+                    Hit = () =>
+                    {
+                        if (AllJudged)
+                        return false;
+
+                        UpdateResult(true);
+                        return true;
+                    },
                 },
+                HeadCircle.CreateProxy(),
```

Comment:
```suggestion
                HeadCircle.CreateProxy(), // required to allow the sliderball to always receive input for judgement purposes
```

should always inline comment when there's a special case. so people know why the special case exists without having to track down git history.